src/libPersistence/BlockStorage.cpp

```diff
@@ -510,6 +591,9 @@ bool BlockStorage::GetAllTxBodiesTmp(std::list<TxnHash>& txnHashes) {
 
 bool BlockStorage::GetAllBlockLink(std::list<BlockLink>& blocklinks) {
   LOG_MARKER();
+
+  shared_lock<shared_timed_mutex> g(m_mutexBlockLink);
+
   leveldb::Iterator* it =
       m_blockLinkDB->GetDB()->NewIterator(leveldb::ReadOptions());
```

Comment:
Maybe let `m_mutexBlockLink` to protect `leveldb::Iterator* it = m_blockLinkDB->GetDB()->NewIterator(leveldb::ReadOptions());` only

Reply:
cannot do that...the iterator is keep changing in the for loop